### PR TITLE
Establish a cats-data project, implement Or

### PR DESCRIPTION
**Pull request by rossabaker**
This is for data types built on cats-core.  Instances of cats-core for std library types belong in std.

### CODE REVIEW

File: data/src/main/scala/cats/data/Or.scala [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+    def reassociateRight[A, B, C](f: Or[Or[A, B], C]) =
+      f.fold(
+        _.fold(
+          Or.left(_),
+          b => Or.right(Or.left(b))
+        ),
+        c => Or.right(Or.right(c))
+      )
+  }
+  */
```
Comment (stew) [inline data/src/main/scala/cats/data/Or.scala]: can we just kill the zombie code?

Comment (rossabaker) [inline data/src/main/scala/cats/data/Or.scala]: I see the following dichotomy of zombies:
1. Depends on types that don't exist yet.
2. Depends on types that won't exist.

I'd like to kill zombie type 2 and leave zombie type 1, if we can distinguish them.  But I'm not opposed to killing all the zombies and gradually adding things back as their dependencies are born.

Comment (stew) [inline data/src/main/scala/cats/data/Or.scala]: sounds good

**Comment by stew** [comment]
:+1:

**Pull request merged**